### PR TITLE
Remove usages of assert throws

### LINKED ISSUE #5320: Remove usages of assertThrows
`org.testng.Assert#assertThrows` encourages very relaxed testing for expected exceptions, which may lead (and has lead) to masking real problems.

Expected exceptions should be tested using `org.assertj.core.api.Assertions#assertThatThrownBy`:
```
assertThatThrownBy(() -> ....)
        .isInstanceOf(IllegalArgumentException.class)
        .hasMessage("Foo is invalid");
```

### PR DESCRIPTION
**Pull request by j4ckofalltrades**
Resolves #5320

### CODE REVIEW

File: presto-thrift-api/src/test/java/io/prestosql/plugin/thrift/api/TestNameValidationUtils.java [hunk 1]
```diff
@@ -15,21 +15,33 @@
...
+        assertInvalidColumnName(() -> checkValidName("ab-c"));
+    }
+
+    private static void assertInvalidColumnName(Supplier<String> columnNameSupplier)
+    {
+        try {
+            columnNameSupplier.get();
+        }
+        catch (Exception e) {
+            assertTrue(e instanceof IllegalArgumentException);
```
Comment (kokosing) [inline presto-thrift-api/src/test/java/io/prestosql/plugin/thrift/api/TestNameValidationUtils.java:44]: Why not assertThatThrownBy?

File: presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/glue/TestHiveGlueMetastore.java [hunk 2]
```diff
@@ -23,7 +23,6 @@
 import io.prestosql.plugin.hive.metastore.HiveMetastore;
 import io.prestosql.plugin.hive.metastore.PartitionWithStatistics;
 import io.prestosql.plugin.hive.metastore.Table;
-import io.prestosql.spi.PrestoException;
```
Comment (kokosing) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/glue/TestHiveGlueMetastore.java]: please squash it with previous commit.

Comment (j4ckofalltrades) [reply presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/glue/TestHiveGlueMetastore.java]: Will do

File: presto-thrift-api/src/test/java/io/prestosql/plugin/thrift/api/TestNameValidationUtils.java (hunk 1 above)
Comment (j4ckofalltrades) [reply presto-thrift-api/src/test/java/io/prestosql/plugin/thrift/api/TestNameValidationUtils.java]: Updated

**Review by aalbu — COMMENTED** [review]
👍

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestBackgroundHiveSplitLoader.java [hunk 3]
```diff
@@ -358,8 +357,10 @@ public void testNoHangIfPartitionIsOffline()
         HiveSplitSource hiveSplitSource = hiveSplitSource(backgroundHiveSplitLoader);
         backgroundHiveSplitLoader.start(hiveSplitSource);
 
-        assertThrows(RuntimeException.class, () -> drain(hiveSplitSource));
-        assertThrows(RuntimeException.class, hiveSplitSource::isFinished);
+        assertThatThrownBy(() -> drain(hiveSplitSource))
+                .isInstanceOf(RuntimeException.class);
```
Comment (findepi) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestBackgroundHiveSplitLoader.java:361]: can we be a bit more specific about exception type?
for example, we should not be OK (test should fail) when the code throws a NullPointerException
here it throws `PrestoException` and we should test for that

similarly, it is desireable to test the exception message
for example no message or some useless message won't be useful for end users.

thus, the asserion here should look like
```
assertThatThrownBy(() -> drain(hiveSplitSource))
        .isInstanceOf(PrestoException.class)
        .hasMessage("OFFLINE");
```

(i know that "OFFLINE" message is far from ideal, but let's keep fixing this as out of scope for the PR)

**Review by findepi — COMMENTED** [review]
Thanks for working on this.

i've added a comment in one place, but the comment is actually general.

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestBackgroundHiveSplitLoader.java (hunk 3 above)
Comment (j4ckofalltrades) [reply presto-hive/src/test/java/io/prestosql/plugin/hive/TestBackgroundHiveSplitLoader.java]: Got it, pushed updates.

Comment (findepi) [reply presto-hive/src/test/java/io/prestosql/plugin/hive/TestBackgroundHiveSplitLoader.java]: Thanks! Would you be able to also add `.hasMessage` / `.hasMessageMatching` in places where we do not check the message at all yet?

Comment (j4ckofalltrades) [reply presto-hive/src/test/java/io/prestosql/plugin/hive/TestBackgroundHiveSplitLoader.java]: Updated

File: presto-main/src/test/java/io/prestosql/security/TestFileBasedSystemAccessControl.java [hunk 4]
```diff
@@ -124,10 +123,16 @@ public void testDocsExample()
...
+        assertThatThrownBy(() -> accessControlManager.checkCanImpersonateUser(Identity.ofUser("alice"), "bob"))
+                .isInstanceOf(AccessDeniedException.class)
+                .hasMessageContaining("Access Denied: User alice cannot impersonate user bob");
+        assertThatThrownBy(() -> accessControlManager.checkCanImpersonateUser(Identity.ofUser("bob"), "alice"))
+                .isInstanceOf(AccessDeniedException.class)
+                .hasMessageContaining("Access Denied: User bob cannot impersonate user alice");
+
+        assertThatThrownBy(() -> accessControlManager.checkCanImpersonateUser(Identity.ofUser("charlie"), "doris"))
+                .isInstanceOf(AccessDeniedException.class)
+                .hasMessageContaining("Access Denied: User charlie cannot impersonate user doris");
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/security/TestFileBasedSystemAccessControl.java:135]: The quality of access control messages is very important from end user perspective.
Thanks for doing this. This is a great testing improvement especially here.

File: presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/glue/TestHiveGlueMetastore.java [hunk 5]
```diff
@@ -227,7 +228,8 @@ public void testGetDatabaseFailureLogsStats()
         GlueHiveMetastore metastore = (GlueHiveMetastore) getMetastoreClient();
         GlueMetastoreStats stats = metastore.getStats();
         long initialFailureCount = stats.getGetDatabase().getTotalFailures().getTotalCount();
-        assertThrows(() -> getMetastoreClient().getDatabase(null));
+        assertThatThrownBy(() -> getMetastoreClient().getDatabase(null))
+                .isInstanceOf(PrestoException.class);
```
Comment (findepi) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/glue/TestHiveGlueMetastore.java:232]: add
```
.hasMessageStartingWith("Database name cannot be equal to null or empty")
```

here

(tested locally)

File: presto-plugin-toolkit/src/test/java/io/prestosql/plugin/base/security/TestFileBasedAccessControl.java [hunk 6]
```diff
@@ -371,7 +370,9 @@ private ConnectorAccessControl createAccessControl(String fileName)
 
     private static void assertDenied(ThrowingRunnable runnable)
     {
-        assertThrows(AccessDeniedException.class, runnable);
+        assertThatThrownBy(runnable::run)
+                .isInstanceOf(AccessDeniedException.class)
+                .hasMessageStartingWith("Access Denied");
```
Comment (findepi) [inline presto-plugin-toolkit/src/test/java/io/prestosql/plugin/base/security/TestFileBasedAccessControl.java:375]: We should improve these tests to match `TestFileBasedSystemAccessControl`

please add a TODO here
```
// TODO test expected message precisely, as in TestFileBasedSystemAccessControl
```

**Comment by findepi** [comment]
Merged as b10bc7fb2dd1db7783762e04c53cfe62e0e9913e.

(i applied the two outstanding minor comments myself)

thanks for doing this!

**Comment by j4ckofalltrades** [comment]
No worries, glad I could help.

**Pull request closed without merging**